extensions/wikia/CreateNewWiki/CreateNewWikiController.class.php

```diff
@@ -274,55 +273,76 @@ public function CreateWiki() {
+		$task->call( 'create', $params['wName'], $params['wDomain'], $params['wLanguage'], $params['wVertical'], $categories, $allAges );
+		$task_id = $task->setQueue( Wikia\Tasks\Queues\PriorityQueue::NAME )->queue();
 
-		try {
-			$createWiki->create();
-		}
-		catch(Exception $ex) {
-			$error_code = $ex->getCode();
-			$this->error(
-				__METHOD__ . ": backend failed to process the request: " . $ex->getMessage(),
-				[
-					'code' => $error_code,
-					'params' => $params,
-					'exception' => $ex
-				]
-			);
-			$this->response->setCode( 500 );
-			$this->response->setVal( self::STATUS_FIELD, self::STATUS_BACKEND_ERROR );
-			$this->response->setVal( self::STATUS_MSG_FIELD, wfMessage( 'cnw-error-general' )->parse() );
-			$this->response->setVal( self::STATUS_HEADER_FIELD, wfMessage( 'cnw-error-general-heading' )->escaped() );
-			$this->response->setVal( self::ERROR_CLASS_FIELD, get_class( $ex ) );
-			$this->response->setVal( self::ERROR_CODE_FIELD, $ex->getCode() );
-			$this->response->setVal( self::ERROR_MESSAGE_FIELD, $ex->getMessage() );
-			wfProfileOut( __METHOD__);
-			return;
-		}
+		// return ID of the created task ID, front-end code will poll its status
+		// e.g {"task_id":"mw-04CF2876-2176-4036-96E5-B6378DB1AC8E"}
+		$this->response->setCode( 201 ); // HTTP 201 Created
+		$this->response->setVal( 'task_id', $task_id );
 
-		$cityId = $createWiki->getCityId();
+		wfProfileOut(__METHOD__);
+	}
 
-		if ( isset($params['wAllAges']) && !empty( $params['wAllAges'] ) ) {
-			WikiFactory::setVarByName( self::WF_WDAC_REVIEW_FLAG_NAME, $cityId, true, __METHOD__ );
-		}
+	/**
+	 * This AJAX entry point is used to poll for wiki creation status
+	 *
+	 * @see SUS-4383
+	 */
+	public function CheckStatus() {
+		global $wgDevelDomains;
 
-		$this->response->setVal( self::STATUS_FIELD, self::STATUS_OK );
-		$this->response->setVal( self::SITE_NAME_FIELD, $createWiki->getSiteName() );
-		$this->response->setVal( self::CITY_ID_FIELD, $cityId );
-		$finishCreateTitle = GlobalTitle::newFromText( "FinishCreate", NS_SPECIAL, $cityId );
+		$task_id = (string) $this->getRequest()->getVal('task_id');
+		$task_details = CreateWikiTask::getCreationLogEntry( $task_id );
 
-		$fullURL = $finishCreateTitle->getFullURL( [ 'editToken' => $wgUser->getEditToken() ] );
-		$finishCreateUrl = empty( $wgDevelDomains ) ? $fullURL : str_replace( '.wikia.com', '.'.$wgDevelDomains[0], $fullURL );
-		$this->response->setVal( 'finishCreateUrl',  $finishCreateUrl );
+		// do not cache, we always want to hit the backend as the response can change anytime
+		$this->response->setCachePolicy( WikiaResponse::CACHE_PRIVATE );
+		$this->response->setCacheValidity( WikiaResponse::CACHE_DISABLED );
 
-		$this->info(__METHOD__ . ': completed', [
-			'city_id' => $cityId,
-			'params' => $params,
-		]);
+		// given task ID not found in the creation log (maybe not yet?)
```

Comment:
We should probably fail at some point, if the task hasn't started (or started and failed to write to the db). We can either create the log record before the task is queued and compare `creation_started` to the current timestamp, or send some information with this request (a counter or start time).

Reply:
Handled in dffd387